src/chains/ethereum/transaction/src/rpc-transaction.ts

```diff
@@ -26,29 +26,29 @@ export type TypedRpcTransaction =
   | EIP2930AccessListRpcTransaction
   | EIP1559FeeMarketRpcTransaction;
 
-export type LegacyRpcTransaction = RpcTransaction & {
-  gasPrice?: string;
-  chainId?: never;
-  accessList?: never;
-  maxPriorityFeePerGas?: never;
-  maxFeePerGas?: never;
+export type LegacyRpcTransaction = Readonly<RpcTransaction> & {
```

Comment:
Curious about the reasoning for this!

Reply:
We should never mutate the json data a user gives us (like in `provider.send("eth_sendTransaction", transaction)`) but in one place in our code we were. Marking the type as `Readonly` prevents that sort of bug from happening.